### PR TITLE
[WIP] scripts: Add a generic test script for running random commands

### PR DESCRIPTION
**Pull request by sboeuf**
This new script is a simple wrapper allowing a user to run a bash script
inside a Cloud-Hypervisor's VM, based on the parameters provided through
the CLI.

Signed-off-by: Sebastien Boeuf <sebastien.boeuf@intel.com>

### CODE REVIEW

File: scripts/generic_tests.sh [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    echo ""
+    echo "Available commands:"
+    echo ""
+    echo "    run --ch --kernel --cmdline --rootfs --cloudinit --guest-user --guest-passwd --mac --ip --mask --guest-ip [--device] --script"
+    echo "        Run the Cloud Hypervisor VFIO generic test."
+    echo "        --ch			Path to the Cloud-Hypervisor binary."
+    echo "        --kernel		Path to the guest kernel."
+    echo "        --cmdline		Guest kernel boot parameters."
+    echo "        --rootfs		Path to the rootfs image."
+    echo "        --cloudinit		Path to the cloudinit image."
```
Comment (sameo) [inline scripts/generic_tests.sh:20]: Can we generate a default cloud-init from the script if that one is empty? And use the passed user/passwd?

File: scripts/generic_tests.sh [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+cmd_help() {
+    echo ""
+    echo "Cloud Hypervisor $(basename $0)"
+    echo "Usage: $(basename $0) <command> [<command args>]"
+    echo ""
+    echo "Available commands:"
+    echo ""
+    echo "    run --ch --kernel --cmdline --rootfs --cloudinit --guest-user --guest-passwd --mac --ip --mask --guest-ip [--device] --script"
```
Comment (sameo) [inline scripts/generic_tests.sh:14]: Out of all those options, which ones are mandatory and which ones are not?

Comment (sameo) [reply scripts/generic_tests.sh:14]: Providing we can generate a default cloud-init image, the mandatory options are: `ch`, `kernel`, `rootfs`

File: scripts/generic_tests.sh [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            ;;
+    esac
+    shift
+done
+
+# $1 is now a command name. Check if it is a valid command and, if so,
+# run it.
+#
+declare -f "cmd_$1" > /dev/null
+ok_or_die "Unknown command: $1. Please use \`$0 help\` for help."
```
Comment (sameo) [inline scripts/generic_tests.sh:112]: Where is that one defined?

**Comment by sboeuf** [comment]
@sameo @rbradford @mcastelino
Ok I think I'm gonna get rid of the script (which I don't really like), but instead please review the commit `ci: Add generic VFIO tests`, I think this should be a pretty simple way to run anything we want.

**Comment by sameo** [comment]
> @sameo @rbradford @mcastelino
> Ok I think I'm gonna get rid of the script (which I don't really like), but instead please review the commit `ci: Add generic VFIO tests`, I think this should be a pretty simple way to run anything we want.

Right, I prefer that approach as well. It's flexible enough and more tightly integrated with our testing framework.

**Comment by rbradford** [comment]
@sboeuf I don't think you mean "random commands"

**Pull request closed without merging**